metrics/metricsreporter.go

```diff
@@ -135,27 +135,24 @@ func (m *MetricsReporter) CaptureRoutesPruned(routesPruned uint64) {
 	m.Batcher.BatchAddCounter("routes_pruned", routesPruned)
 }
 
-func (m *MetricsReporter) CaptureRegistryMessage(msg ComponentTagged) {
+func (m *MetricsReporter) CaptureRegistryMessage(msg ComponentTagged, action string) {
 	var componentName string
 	if msg.Component() == "" {
-		componentName = "registry_message"
+		componentName = "registry_message." + action
 	} else {
-		componentName = "registry_message." + msg.Component()
+		componentName = "registry_message." + action + "." + msg.Component()
 	}
 	m.Batcher.BatchIncrementCounter(componentName)
 }
 
-func (m *MetricsReporter) CaptureUnregistryMessage(msg ComponentTagged) {
+func (m *MetricsReporter) CaptureUnregistryMessage(msg ComponentTagged, action string) {
 	var componentName string
 	if msg.Component() == "" {
-		componentName = "unregistry_message"
+		componentName = "unregistry_message." + action
```

Comment:
suggestion to reverse the logic a bit. This suggestion can't be applied all at once because I can't reference the necessary lines in this PR.
```suggestion
componentName := "unregisty_message." + action
if msg.Component() != "" {
    componentName = componentName + "." msg.Component()
}
```

Reply:
Applied